tools/lib/contracts.sh

```
@@ -0,0 +1,386 @@
+        log_error "Try using a proxy pattern or optimizing the contract"
+        return 1
+    fi
+
+    return 0
+}
+
+# Deploy contract with forge create
+deploy_contract() {
+    local sol_file="$1"
+    local contract_name="$2"
+    local target_address="$3"
```

Shellcheck (0.10.0)

[warning] target_address appears unused. Verify use (or export if used externally).

(SC2034)